### PR TITLE
Riedl

### PR DESCRIPTION
**Pull request by jriedl**

### CODE REVIEW

File: cronjobs/zjs_tag_relbib.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    zotero_journal_status_issues = github_api_util.GetAllIssuesForUBTueRepository(ZOTERO_JOURNAL_STATUS_REPO)
+    for issue in zotero_journal_status_issues:
+        issn_matcher = github_api_util.GetISSNMatcher()
+        issns = issn_matcher.findall(issue['title'])
+        if len(issns):
+            for issn in issns:
+                ssgn = GetSSGNStatusFromConfig(config, issn)
+                has_relbib_ssgn = False
+                if ssgn == "FG_0":
+                    #print("SSGN for " + issue['title'] + ": " + ssgn)
```
Comment (mtrojan-ub) [inline cronjobs/zjs_tag_relbib.py:34]: Maybe remove this debug statement?

File: java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java [hunk 2]
```diff
@@ -1615,12 +1615,57 @@ public String getSubfieldBasedSeparator(Map<String, String> separators, char sub
...
+    Predicate<DataField> _LOK689IsTimeSubject = (DataField marcField) -> {
+        if (!(marcField.getTag().equals("LOK") && marcField.getSubfield('0') != null &&
+              marcField.getSubfield('0').getData().substring(0,3).equals("689")))
             return true;
-        Subfield subfieldQ = marcField.getSubfield('q');
-        Subfield subfieldD = marcField.getSubfield('d');
-        return (subfieldQ != null && subfieldQ.getData().equals("z")) || (subfieldD != null && subfieldD.getData().equals("z"));
+        Subfield subfield0 = marcField.getSubfield('0');
+        if (subfield0 != null && subfield0.getData().substring(0,3).equals("689")) {
+            Subfield subfieldA = marcField.getSubfield('a'); //Should be capital
```
[the remark this answers is not in the document]
Comment (jriedl) [reply java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java:1626]: I intentionally put them in. In fact, the subfield in questions is 'A'. However, It was not possible to get `org.marc4j.MarcPermissiveStreamReader.upperCaseSubfields` to work, so this is a workaround to skip the one letter categorizations in these fields.

File: java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java [hunk 3]
```diff
@@ -1615,12 +1615,57 @@ public String getSubfieldBasedSeparator(Map<String, String> separators, char sub
      * In the q-Subfield of 689 standardized subjects we have "z" (time subject), "f" (genre subject), "g" (region subject)
      * Alternatively this is in the d-Subfields
      */
-    Predicate<DataField> _689IsTimeSubject = (DataField marcField) -> {
-        if (!marcField.getTag().equals("689"))
+
+
+    Predicate<DataField> _LOK689IsTimeSubject = (DataField marcField) -> {
+        if (!(marcField.getTag().equals("LOK") && marcField.getSubfield('0') != null &&
```
Comment (mtrojan-ub) [inline java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java:1621]: `(!(marcField.getTag().equals("LOK") && marcField.getSubfield('0') != null && marcField.getSubfield('0').getData().substring(0,3).equals("689")))`

This expression seems to be very long and is used multiple times. Can we define a shortcut for this, e.g. _IsLOKTag("689")?

File: java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java:1626]: In that case the comment should also be added to 1665 as well for consistency reasons

**Pull request merged**